spec/services/atmosphere/billing_service_spec.rb

```diff
@@ -255,15 +330,20 @@
       # Standard non-shared appliance
       appl1 = create(:appliance, fund: cs_fund, appliance_set: wf, appliance_type: not_shareable_appl_type, appliance_configuration_instance: config_inst1, virtual_machines: [not_shareable_vm1])
 
-      cost_unit = appl1.virtual_machines.first.virtual_machine_flavor.hourly_cost # Cost per 1h of use for a single (full) VM
+      vmf = appl1.virtual_machines.first.virtual_machine_flavor
+      vmf.set_hourly_cost_for(Atmosphere::OSFamily.first, 10)
+      vmf.reload
+
+      cost_unit = appl1.virtual_machines.first.virtual_machine_flavor.virtual_machine_flavor_os_families.first.hourly_cost # Cost per 1h of use for a single (full) VM
 
       Atmosphere::BillingService.bill_appliance(appl1, Time.now.utc, "mock billing", true)
 
       cs_fund.reload
       expect(cs_fund.balance).to eq original_balance - cost_unit
 
+
       # Advance the clock by 1.5 hours and destroy appl1
-      appl1.prepaid_until = Time.now - 30.minutes
+      appl1.deployments.first.prepaid_until = Time.now - 30.minutes
       appl1.save
 
       begin
```

Comment:
This rescue is suspicious - conflict exception should not be thrown by fog while destroying vm